compiler/src/yul/mappers/expressions.rs

```diff
@@ -38,11 +38,33 @@ pub fn expr(context: &Context, exp: &Spanned<fe::Expr>) -> Result<yul::Expressio
     }
 }
 
-pub fn expr_call(exp: &Spanned<fe::Expr>) -> Result<yul::Expression, CompileError> {
-    if let fe::Expr::Call { args: _, func } = &exp.node {
+pub fn expr_call(
+    context: &Context,
+    exp: &Spanned<fe::Expr>,
+) -> Result<yul::Expression, CompileError> {
+    if let fe::Expr::Call { args, func } = &exp.node {
         if let fe::Expr::Attribute { value: _, attr } = &func.node {
+            let arguments = &args.node;
+            let yul_args = arguments
+                .iter()
+                .map(|val| {
+                    if let fe::CallArg::Arg(arg_expr) = val.node.clone() {
+                        let spanned = Spanned {
+                            node: arg_expr,
+                            span: val.span,
+                        };
+                        return expr(context, &spanned);
+                    }
+                    unimplemented!()
+                })
+                .filter_map(Result::ok);
+
             let func_name = identifier! { (attr.node) };
-            return Ok(expression! { [func_name]() });
+
+            return Ok(yul::Expression::FunctionCall(yul::FunctionCall {
+                identifier: func_name,
+                arguments: yul_args.collect(),
+            }));
```

Comment:
@g-r-a-n-t I bypassed the macro here. There's probably another way to do that?

Reply:
yup! `expression! { [func_name]([yul_args...]) }` should do the trick

Reply:
Beautiful :sweat_smile: